### PR TITLE
feat: add fromDebtRegistryEntry endpoint to adapter

### PR DESCRIPTION
**Pull request by nadavhollander**
This PR introduces the following changes:

- Adds a `fromDebtRegistryEntry` endpoint to the SimpleInterestLoanAdapter.
- Adds tests for the new endpoint.

### CODE REVIEW

File: src/apis/contracts_api.ts [hunk 1]
```diff
@@ -47,14 +47,15 @@ export const ContractsError = {
                 address ${principalToken}`,
     CANNOT_FIND_TOKEN_WITH_SYMBOL: (symbol: string) =>
         singleLineString`Could not find token associated with symbol ${symbol}.`,
+    CANNOT_FIND_TOKEN_WITH_INDEX: (index: number) =>
+        singleLineString`Could not find token associated with indedx ${index}.`,
```
Comment (graemecode) [inline src/apis/contracts_api.ts]: typo: `indedx`

File: src/adapters/simple_interest_loan_adapter.ts [hunk 2]
```diff
@@ -80,6 +80,10 @@ export const SimpleInterestAdapterErrors = {
         singleLineString`Terms contract parameters are invalid for the given debt order.
                          Principal token at address ${principalTokenAddress} does not
                          correspond to specified token with symbol ${symbol}`,
+    MISMATCHED_TERMS_CONTRACT: (termsContract: string) =>
+        singleLineString`Terms contract at address ${termsContract} is not
+                         a SimpleInterestTermsContract.  As such, this adapter will not
+                         interface with the terms contract as expected.`,
```
Comment (graemecode) [inline src/adapters/simple_interest_loan_adapter.ts]: Not a big deal, but there is actually a grammar inconsistency here - see the above error message does not end in a period.

File: src/adapters/simple_interest_loan_adapter.ts [hunk 3]
```diff
@@ -330,6 +334,37 @@ export class SimpleInterestLoanAdapter {
...
+            interestRate,
+            termLength,
+            amortizationUnit,
+        } = this.termsContractInterface.unpackParameters(entry.termsContractParameters);
+
+        const principalTokenSymbol = await this.contracts.getTokenSymbolByIndexAsync(
+            principalTokenIndex,
+        );
+
+        return {
```
Comment (saturnial) [inline src/adapters/simple_interest_loan_adapter.ts]: Nit, but would be helpful to the reader to construct the specific object that is being returned here: SimpleInterestLoanOrder. Explicit over implicit.

File: src/adapters/simple_interest_loan_adapter.ts [hunk 4]
```diff
@@ -356,4 +391,12 @@ export class SimpleInterestLoanAdapter {
             );
         }
     }
+
+    private async assertIsSimpleInterestTermsContract(termsContract: string): Promise<void> {
```
Comment (saturnial) [inline src/adapters/simple_interest_loan_adapter.ts]: `termsContractAddress`

File: src/apis/contracts_api.ts [hunk 5]
```diff
@@ -364,7 +365,13 @@ export class ContractsAPI {
     public async getTokenSymbolByIndexAsync(index: BigNumber): Promise<string> {
         const tokenRegistryContract = await this.loadTokenRegistry();
 
-        return tokenRegistryContract.getTokenSymbolByIndex.callAsync(index);
+        const symbol = await tokenRegistryContract.getTokenSymbolByIndex.callAsync(index);
+
+        if (symbol === "") {
```
Comment (saturnial) [inline src/apis/contracts_api.ts]: any case where this comes back undefined or null? because those wouldn't throw.

Comment (nadavhollander) [reply src/apis/contracts_api.ts]: I'll add those cases just in case.

**Pull request merged**